subsys/logging/Kconfig.backends

```diff
@@ -325,6 +325,48 @@ source "subsys/logging/Kconfig.template.log_format_config"
 
 endif # LOG_BACKEND_ADSP
 
+config LOG_BACKEND_CAVS_HDA
+	bool "cAVS HDA backend"
+	depends on SOC_FAMILY_INTEL_ADSP && DMA && DMA_CAVS_HDA
+	help
+	  Provide a logging backend which writes to a buffer and
+	  periodically flushes to hardware using ringbuffer like
+	  semantics provided by DMA_CAVS_HDA.
+
+if LOG_BACKEND_CAVS_HDA
+
+backend = CAVS_HDA
+backend-str = cavs_hda
+source "subsys/logging/Kconfig.template.log_format_config"
+
+config LOG_BACKEND_CAVS_HDA_SIZE
+	int "Size of ring buffer"
+	range 128 8192
+	default 4096
+	help
+	  Sets the ring buffer size cAVS HDA uses for logging. Effectively
+	  determines how many log messages may be written to in a period of time.
+	  The period of time is decided by how often to inform the hardware of
+	  writes to the buffer.
+
+config LOG_BACKEND_CAVS_HDA_CHANNEL
```

Comment:
Can we setup hda channel in runtime?
In SOF host will send IPC with channel info

Reply:
In SOF, we currently allow lazy alloc of the HDA DMA backend. Tracing can be started, but as the DMA channel is not yet known, nothing is sent out. This is nice to gather very early logs in the buffers. At one point in boot, the host side kernel sends IPC that configures DMA trace and passes the channel. This comes to SOF in: https://github.com/thesofproject/sof/blob/main/src/ipc/ipc3/handler.c#L796
And ends up configuring the DMA here:
https://github.com/thesofproject/sof/blob/main/src/trace/dma-trace.c#L463

Something similar would be nice for this backend as well.

Reserving a fixed channel is problematic as 1) at least in theory. we could have usages where we want all DMA channels for non-log use, 2) we have large number of old Linux kernel versions deployed that expect SOF FW ABI to allow to dynamically configure the DMA trace/log channel. Requiring a fix channel would break all these old kernels and getting a kernel change deployed (to all distros and users) is much slower and complicated than distributing a new FW. So in short, ABI compatibility is sacred in SOF.

Reply:
Fixed, the logger is now autostarted and fills the buffer with log messages with a "keeps oldest" as the behavior, it now requires a channel param be passed in